mmv1/third_party/terraform/verify/validation.go

```diff
@@ -22,7 +22,7 @@ const (
 
 	SubnetworkLinkRegex = "projects/(" + ProjectRegex + ")/regions/(" + RegionRegex + ")/subnetworks/(" + SubnetworkRegex + ")$"
 
-	RFC1035NameTemplate = "[a-z](?:[-a-z0-9]{%d,%d}[a-z0-9])"
+	RFC1035NameTemplate = "[a-z]([-a-z0-9]{%d,%d}[a-z0-9])?"
```

Comment:
This allows the string "a" when any input is supplied. For example, `1,61` (source input to the fn of `2,63`). Added some extra test cases as constraints.

I suspect we need to rework the regex substantially to go below a minimum of 2 while properly validating the format. We may want to mix procedural code and regex or just write this as entirely procedural code.

Reply:
agreed. After checking this regex fails on one new test case that you've added. This is mainly due to length which should be procedural imo.

The only other think that this regex is affecting are Service Account names so will look out for errors there as well.